In [ ]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

pd.options.display.max_columns = 100


print('numpy version : ',np.__version__)
print('pandas version : ',pd.__version__)
print('seaborn version : ',sns.__version__)

numpy version :  1.25.2
pandas version :  2.0.3
seaborn version :  0.13.1


## Load Dataset

In [ ]:
# Load dataset in Google Colab
data_path = '/content/superstore_dataset.csv'
data = pd.read_csv(data_path)

---
Data Understanding

---



In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 19 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   order_id       9994 non-null   object 
 1   order_date     9994 non-null   object 
 2   ship_date      9994 non-null   object 
 3   customer       9994 non-null   object 
 4   manufactory    9994 non-null   object 
 5   product_name   9994 non-null   object 
 6   segment        9994 non-null   object 
 7   category       9994 non-null   object 
 8   subcategory    9994 non-null   object 
 9   region         9994 non-null   object 
 10  zip            9994 non-null   int64  
 11  city           9994 non-null   object 
 12  state          9994 non-null   object 
 13  country        9994 non-null   object 
 14  discount       9994 non-null   float64
 15  profit         9994 non-null   float64
 16  quantity       9994 non-null   int64  
 17  sales          9994 non-null   float64
 18  profit_m

Berdasarkan output diatas, dapat disimpulkan bahwa dataset tersebut, memiliki total: 19 kolom, dengan jumlah maksimal baris untuk setiap kolom sebanyak: 9994 baris.

# Data Cleaning and Preparation

---
Time Series

---




In [ ]:
data['order_date'] = pd.to_datetime(data['order_date'])
data['ship_date'] = pd.to_datetime(data['ship_date'])

#extract year and month from order_date
data['order_year'] = data['order_date'].dt.year
data['order_month'] = data['order_date'].dt.month

#extract year and month from ship_date
data['ship_year']=data['ship_date'].dt.year
data['ship_month']=data['ship_date'].dt.month

data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 23 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   order_id       9994 non-null   object        
 1   order_date     9994 non-null   datetime64[ns]
 2   ship_date      9994 non-null   datetime64[ns]
 3   customer       9994 non-null   object        
 4   manufactory    9994 non-null   object        
 5   product_name   9994 non-null   object        
 6   segment        9994 non-null   object        
 7   category       9994 non-null   object        
 8   subcategory    9994 non-null   object        
 9   region         9994 non-null   object        
 10  zip            9994 non-null   int64         
 11  city           9994 non-null   object        
 12  state          9994 non-null   object        
 13  country        9994 non-null   object        
 14  discount       9994 non-null   float64       
 15  profit         9994 n

---
**Missing Value**

---



Untuk mendeteksi suatu Missing Value dapat digunakan fungsi isnull()

In [ ]:
data.isnull().any()

order_id         False
order_date       False
ship_date        False
customer         False
manufactory      False
product_name     False
segment          False
category         False
subcategory      False
region           False
zip              False
city             False
state            False
country          False
discount         False
profit           False
quantity         False
sales            False
profit_margin    False
dtype: bool

Berdasarkan output diatas, jika hasil output tertulis False, artinya kolom tersebut tidak mengandung Missing Values.
Untuk mengetahui jumlah missing value tiap kolom, dapat digunakan sum()

In [ ]:
data.isnull().sum()

order_id         0
order_date       0
ship_date        0
customer         0
manufactory      0
product_name     0
segment          0
category         0
subcategory      0
region           0
zip              0
city             0
state            0
country          0
discount         0
profit           0
quantity         0
sales            0
profit_margin    0
dtype: int64

Berdasarkan output summarize tersebut, tidak kolom yang mengandung Missing Values (blanks/ N/A (Python : nan atau NaN))

In [ ]:
#Total Number of Missing NA
data.isnull().sum().sum()

0

# Data Understanding

In [ ]:
data.describe(exclude=['int64', 'float64', 'datetime64[ns]', 'int32']).transpose()

count unique              top  freq
order_id      9994   5009   US-2023-100111    14
customer      9994    793    William Brown    37
manufactory   9994    182            Other  1893
product_name  9994   1849  Staple envelope    48
segment       9994      3         Consumer  5191
category      9994      3  Office Supplies  6026
subcategory   9994     17          Binders  1523
region        9994      4             West  3203
city          9994    531    New York City   915
state         9994     49       California  2001
country       9994      1    United States  9994

Dari output di atas, didapatkan beberapa informasi, diantaranya:
1.  order_id merupakan unique identifier yang nanti akan digunakan untuk menghitung rows di Tableau serta menjadi primary ket di SQL table creation
2. terdapat 793 customer unik, bisa jadi karena order di hari yg sama namun berbeda barang. Customer paling sering melakukan transaksi ialah William Brown sebanyak 37 kali
3. 1849 macam produk dalam toko ini. Produk yang paling banyak dibeli ialah Staple envelope
4. New York City menjadi top city dari seluruh 531 kota di United States


In [ ]:
cats_to_visualize=['product_name', 'category', 'subcategory', 'city', 'state']

In [ ]:
#Categorical Counts
for col in cats_to_visualize :
  print(f'Value counts untuk {col} :')
  print(data[col].value_counts(), '\n')

Value counts untuk product_name :
product_name
Staple envelope                                                      48
Staples                                                              46
Easy-staple paper                                                    46
Avery Non-Stick Binders                                              20
Staples in misc. colors                                              19
                                                                     ..
Eldon Jumbo ProFile Portable File Boxes Graphite/Black                1
Newell 342                                                            1
Belkin 7 Outlet SurgeMaster Surge Protector with Phone Protection     1
RCA ViSYS 25423RE1 Corded phone                                       1
Acco Glide Clips                                                      1
Name: count, Length: 1849, dtype: int64 

Value counts untuk category :
category
Office Supplies    6026
Furniture          2121
Technology         1847
Name: co

In [ ]:
#Export the final dataframe
data.to_csv('superstore_dataset_final.csv', index=False)